lightningd/dual_open_control.c

```diff
@@ -3085,15 +3073,8 @@ static struct command_result *json_queryrates(struct command *cmd,
 		return command_fail(cmd, FUNDING_PEER_NOT_CONNECTED,
 				    "Peer not connected");
 
-	/* We can't query rates for a peer we have a channel with */
-	channel = peer_active_channel(peer);
-	if (channel)
-		return command_fail(cmd, LIGHTNINGD, "Peer in state %s,"
-				    " can't query peer's rates if already"
-				    " have a channel",
-				    channel_state_name(channel));
-
-	channel = peer_unsaved_channel(peer);
+	/* FIXME: This is wrong! */
```

Comment:
I don't quite follow why this would be wrong.

Reply:
We really shouldn't be re-using an unsaved channel here: we should *always* create a new one I think.